ExploreASL_Initialize.m

```diff
@@ -697,111 +639,242 @@
+                        fprintf(', codes for two-sample data');
+                elseif  x.S.Sets1_2Sample(iSet)==3
+                        fprintf([', continuous variate (with ' num2str(length(unique(x.S.SetsID(:,iSet)))) ' unique values)']);
+                end                    
+
+                fprintf('\n');
+            end
+    else    
+        fprintf('%s\n','No sets are defined');
+    end
+
+    if ~isfield(x,'M0')
+    %     warning('M0 option missing!');
+    else
+        fprintf('\n%s\n',['M0 option selected is "' num2str(x.M0) '"']);
+    end
+
+    fprintf('\nx.D.ROOT\t\t\t\t%s\n',x.D.ROOT);
+    fprintf('x.DELETETEMP\t\t\t%s\n',[num2str(x.DELETETEMP) ' (delete temporary files)']);
+    fprintf('x.Quality\t\t\t\t%s\n',[num2str(x.Quality) ' (0 = fast try-out; 1 = normal high quality)']);
+
+
+    %% -----------------------------------------------------------------------
+    %% 3) Print warnings
+    fprintf('\n%s\n\n','==============================================================================================');
+    field_symbol = {'subject_regexp'};
+
+    for iField=1:length(field_symbol)
+        if ~isfield(x,field_symbol{iField})
+            warning(['x.' field_symbol{iField} ' was not defined in DATA_PAR.m!'])
+        end
+    end
+
+    if ~isfield(x,'D')
+        warning('x.D didn''nt exist');
+    else
+        field_symbol = {'ROOT'};
+        for iField=1:length(field_symbol)
+            if ~isfield(x.D,field_symbol{iField})
+                warning(['x.D.' field_symbol{iField} ' was not defined in DATA_PAR.m!'])
             end
-            if      x.S.Sets1_2Sample(iSet)==1
-                    fprintf(', codes for paired data');
-            elseif  x.S.Sets1_2Sample(iSet)==2
-                    fprintf(', codes for two-sample data');
-            elseif  x.S.Sets1_2Sample(iSet)==3
-                    fprintf([', continuous variate (with ' num2str(length(unique(x.S.SetsID(:,iSet)))) ' unique values)']);
-            end                    
-
-            fprintf('\n');
         end
-else    
-    fprintf('%s\n','No sets are defined');
-end
+    end
+
+    if ~isempty(regexp(x.subject_regexp, '^(\^|)\.\*(\$|)$'))
+        warning('Subject regexp not specific! Check that no wrong folders are included as subjects');
+    end
+
+    fprintf('\n');
 
-if ~isfield(x,'M0')
-%     warning('M0 option missing!');
-else
-    fprintf('\n%s\n',['M0 option selected is "' num2str(x.M0) '"']);
 end
 
-fprintf('%s\n',['x.DELETETEMP = ' num2str(x.DELETETEMP) ' (delete temporary files)']);
-fprintf('%s\n',['x.Quality    = ' num2str(x.Quality) ' (0 = fast try-out; 1 = normal high quality)']);
+%% -----------------------------------------------------------------------
+%% Add ExploreASL Directory
+function addExploreASLDirectory(MyPath)
 
+    % Define paths (should be equal when loading data or only initializing)
+    addpath(MyPath); % ExploreASL
+    subfolders_to_add = {'Functions', 'mex', 'Modules', 'Testing',...
```

Comment:
perhaps good to order these alphabetically, easier to check

Reply:
Fixed notation to camel case and improved the sorting in 7762591.